saleor/payment/gateways/adyen/plugin.py

```diff
@@ -242,16 +247,37 @@ def process_payment(
             kind = TransactionKind.CAPTURE
         else:
             kind = TransactionKind.AUTH
+
+        action = result.message.get("action")
```

Comment:
I would suggest extracting `action` logic to separate function. Process_payment is growing and adding helper function for `action` would improve readability.